engine/evd.ml

```diff
@@ -622,11 +622,14 @@ let merge_universe_context evd uctx' =
 let set_universe_context evd uctx' =
   { evd with universes = uctx' }
 
-(* TODO: make unique *)
+let eq_conv_pb (pbty,env,t1,t2) (pbty',env',t1',t2') =
+  pbty == pbty' && env == env' && Constr.equal t1 t1' && Constr.equal t2 t2'
+
```

Comment:
Are you sure you shouldn't be evar-insensitive here?

Reply:
Good point. I made a wrapper in evarutil taking a unification pb with EConstr's, this also avoids EConstr.Unsafe.to_constr's in Evarconv.